core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -196,12 +206,24 @@ public DefaultMuleMessage(Object message, MuleMessage previous, MuleContext mule
         if (message instanceof MuleMessage)
         {
             MuleMessage payloadMessage = (MuleMessage) message;
+            dataType = createDataTypeWrapper(payloadMessage.getDataType());
             setPayload(payloadMessage.getPayload());
             copyMessageProperties(payloadMessage);
         }
         else
         {
-            setPayload(message);
+            DataType newDataType;
+            if (isCloningMessage)
```

Comment:
Wondering if there is another way to do this to avoid flag and if/else.  Could this constrcutor recieve the dataType maybe?